google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/ManagedVmCloudConfigurable.java

```diff
@@ -42,31 +46,63 @@
 public class ManagedVmCloudConfigurable extends RemoteServerConfigurable implements Configurable {
 
   private final ManagedVmServerConfiguration configuration;
-  @Nullable
-  private final Project project;
+  private final SystemEnvironmentProvider environmentProvider;
 
   private String displayName = GctBundle.message("appengine.managedvm.name");
   private JPanel myMainPanel;
-  private TextFieldWithBrowseButton cloudSdkLocationField;
+  private TextFieldWithBrowseButton cloudSdkDirectoryField;
   private ProjectSelector projectSelector;
+  private JLabel warningMessage;
 
   public ManagedVmCloudConfigurable(ManagedVmServerConfiguration configuration,
       @Nullable Project project) {
     this.configuration = configuration;
-    this.project = project;
+    environmentProvider = SystemEnvironmentProvider.getInstance();
+
+    warningMessage.setVisible(false);
+
+    final String cloudSdkExecutablePath = CloudSdkUtil.findCloudSdkExecutablePath(environmentProvider);
+    final String cloudSdkDirectoryPath = CloudSdkUtil.findCloudSdkDirectoryPath(environmentProvider);
 
-    String cloudSdkPath = CloudSdkUtil.findCloudSdkPath();
-    if (cloudSdkPath != null && configuration.getCloudSdkPath() == null) {
-      configuration.setCloudSdkPath(cloudSdkPath);
-      cloudSdkLocationField.setText(cloudSdkPath);
+    if (cloudSdkExecutablePath != null
+        && cloudSdkDirectoryPath != null
+        && configuration.getCloudSdkExecutablePath() == null) {
+      configuration.setCloudSdkExecutablePath(cloudSdkExecutablePath);
+      cloudSdkDirectoryField.setText(cloudSdkDirectoryPath);
     }
-    cloudSdkLocationField.addBrowseFolderListener(
+
+    cloudSdkDirectoryField.addBrowseFolderListener(
         GctBundle.message("appengine.cloudsdk.location.browse.window.title"),
         null,
         project,
-        FileChooserDescriptorFactory.createSingleFileDescriptor()
+        FileChooserDescriptorFactory.createSingleFolderDescriptor()
     );
-    projectSelector.getDocument().addDocumentListener(new DocumentAdapter() {
+
+    cloudSdkDirectoryField.getTextField().getDocument()
+        .addDocumentListener(getSdkDirectoryFieldListener());
+    projectSelector.getDocument().addDocumentListener(getProjectSelectorListener());
+  }
+
+  private DocumentAdapter getSdkDirectoryFieldListener() {
+    return new DocumentAdapter() {
+      @Override
+      protected void textChanged(DocumentEvent e) {
+        String path = cloudSdkDirectoryField.getText();
+        boolean isValid = CloudSdkUtil.containsCloudSdkExecutable(path);
+        if(!isValid) {
```

Comment:
flipping the if branches could improve readability a bit by not needing the negation